### PR TITLE
Collect dependencies as separate code areas.

### PR DESCRIPTION
**Pull request by goto-bus-stop**
This still needs some styling work (a long dependency list overflows the page), probably more

Sample here:
https://upload.clinicjs.org/public/9bb3f6bd3acd7178b5cb8faa9590f6796859631213561522ba0b9c1391b67d59/5338.clinic-flame.html

Try toggling `async` or `bunyan` for some highly visible examples.

### CODE REVIEW

**Comment by AlanSl** [comment]
Thinking about the best way to proceed with this - can you add a test and, in this PR, have it not display anything in the options menu (maybe even have a line in the options menu component that skips types where category is deps, that we remove in the PR that updates the UI for this).

Then we can review and release it, and do the UI side separately, without getting into too much of a PR chain.

**Comment by goto-bus-stop** [comment]
I think the easiest way to approach that would be by commenting some code on the analysis side and clearing the list of dependency code areas before returning it to the visualizer. I tried doing it in the UI only but since this changes the `excludeKey` for dependencies it'd require changes in different parts of the UI to keep the old behaviour.

Working on some tests now!

File: analysis/code-areas.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+// Turn a Set of code area names into a sorted list of code area objects.
```
Comment (AlanSl) [inline analysis/code-areas.js]: Missing `'use strict'`

File: visualizer/ui.js [hunk 2]
```diff
@@ -378,8 +378,21 @@ class Ui extends events.EventEmitter {
       'all-v8:cpp': 'V8 C++',
       'all-v8:regexp': 'RegExp'
     }
+
+    if (keysToLabels[key]) {
+      return keysToLabels[key]
+    }
+
+    if (key.startsWith('deps:')) {
```
Comment (AlanSl) [inline visualizer/ui.js]: This is doing something very similar to the logic below, maybe we could replace this and that with something like 

```
    const splitKey = key.split(':')
    if (splitKey.length > 1) {
      const presetLabel = keysToLabels[splitKey[1]]
      return presetLabel || splitKey[1]
    }
```

...which will then just work if we add different types to app again.

Also fixes a mistake that slipped in to the previous code as well, `.split()`'s array is always length > 0 (and always truthy, which was a mistake before).

File: test/analysis-dependencies.test.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
+const { test } = require('tap')
```
Comment (AlanSl) [inline test/analysis-dependencies.test.js]: Also missing `'use strict'`

**Comment by goto-bus-stop** [comment]
Actually I should address the hash issue here. history.js has a hardcoded list of exclusions that should become dynamic with this patch

**Comment by AlanSl** [comment]
What's the expected behaviour if there are no dependencies?

For example I just tried it on `./test/fixtures/inspect.js` - current behaviour seems fine to me:

- With `removeDependencyAreasFromCodeAreas` active, no change, there's a Dependencies tick box that ticks and unticks without doing anything except changing the hash
- With `removeDependencyAreasFromCodeAreas` disabled, there's a Dependencies tick box that is unticked and can't be ticked, but doesn't cause any errors. https://upload.clinicjs.org/public/97f850be4d398e2333ceb52e8dfd2ca9263608952d3349766e9fde0dd1a6d439/32119.clinic-flame.html

...then we can add a greying-out disabled property or dynamic generation to components that need them. 

So, seems fine, just checking this behaviour isn't a side effect of something that might have other side effects I'm not seeing.

**Comment by goto-bus-stop** [comment]
That would still be one frame of app code, I think (the anonymous function wrapper that Node adds around each file)

There will always be Node.js in the init frames, but doing some pure logic in a `while` loop or something may work!

**Comment by AlanSl** [comment]
How much are we expecting to work from this PR currently if `removeDependencyAreasFromCodeAreas` is skipped? For me, all the tickboxes update the graph, but neither individual dependencies nor now the overall Dependencies tickbox are updating the hash currently. 

Or is that work in progress? 

Example: https://upload.clinicjs.org/public/8b48303f57367d8eaf5b0c30ce5941c9fab12d73b7819f809a30333ef0c57c9d/2010.clinic-flame.html#selectedNode=198&zoomedNode=&exclude=13e&merged=true

**Comment by goto-bus-stop** [comment]
On a profile with dependencies, both work for me locally, with latest pull from this branch. That link appears to be generated from https://github.com/nearform/node-clinic-flame/pull/62/commits/ec3d9a010931d05b87413774bc19337f88598149 which doesn't include the latest hash changes.

**Comment by AlanSl** [comment]
You're right, last pull failed and I hadn't noticed. All behaves as expected with the lastest commits, LGTM

**Pull request merged**